sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -784,25 +920,31 @@ class GpuMergeAggregateIterator(
         }
       }
 
-      if (aggregatedBatches.size() > 1) {
+      if (aggregatedBatches.size > 1) {
         // Unable to merge to a single output, so must fall back
         if (allowNonFullyAggregatedOutput && shouldSkipThirdPassAgg) {
           // skip third pass agg, return the aggregated batches directly
           logInfo(s"Rows after second pass aggregation exceeds " +
             s"${skipAggPassReductionRatio * 100}% of " +
             s"rows after first pass, skip the third pass agg")
           fallbackIter = Some(new Iterator[ColumnarBatch] {
-            override def hasNext: Boolean = !aggregatedBatches.isEmpty
+            override def hasNext: Boolean = aggregatedBatches.nonEmpty
 
             override def next(): ColumnarBatch = {
-              withResource(aggregatedBatches.pop()) { spillableBatch =>
+              withResource(aggregatedBatches.remove(0)) { spillableBatch =>
                 spillableBatch.getColumnarBatch()
               }
             }
           })
         } else {
           // fallback to sort agg, this is the third pass agg
-          fallbackIter = Some(buildSortFallbackIterator())
+          aggFallbackAlgorithm.toLowerCase match {
```

Comment:
nit: Could we use an enum or something like it here? a string comparison feels potentially problematic.